src/jit/emitxarch.cpp

```diff
@@ -90,22 +99,45 @@ bool emitter::IsThreeOperandMoveAVXInstruction(instruction ins)
     return IsAVXInstruction(ins) &&
            (ins == INS_movlpd || ins == INS_movlps || ins == INS_movhpd || ins == INS_movhps || ins == INS_movss);
 }
-#endif // FEATURE_AVX_SUPPORT
 
-// Returns true if the AVX instruction is a 4-byte opcode.
+// ------------------------------------------------------------------------------
+// Is4ByteAVXInstruction: Returns true if the AVX instruction is a 4-byte opcode.
+//
+// Arguments:
+//    ins  -  instructions
+//
 // Note that this should be true for any of the instructions in instrsXArch.h
 // that use the SSE38 or SSE3A macro.
+//
 // TODO-XArch-Cleanup: This is a temporary solution for now. Eventually this
 // needs to be addressed by expanding instruction encodings.
-bool Is4ByteAVXInstruction(instruction ins)
+bool emitter::Is4ByteAVXInstruction(instruction ins)
 {
-#ifdef FEATURE_AVX_SUPPORT
-    return (ins == INS_dpps || ins == INS_dppd || ins == INS_insertps || ins == INS_pcmpeqq || ins == INS_pcmpgtq ||
+    return UseAVX() &&
```

Comment:
Some of emitter functionality is gated on Is4ByteAVXInstruction(ins) that wasn't checking whether AVX is available for use.  Those code paths should not execute when SSE3_4 is available and AVX is not.  Earlier it was not needed because these instructions are emitted only in AVX mode.  Now post this change, there are some SSE3_4 instructions that are common with AVX.